src/VirtualClient/VirtualClient.Actions/SysbenchOLTP/SysbenchOLTPClientExecutor.cs

```diff
@@ -0,0 +1,319 @@
+            this.InitializeApiClients();
+
+            // get sysbench workload path
+            DependencyPath workloadPackage = await this.packageManager.GetPlatformSpecificPackageAsync(this.PackageName, this.Platform, this.CpuArchitecture, CancellationToken.None).ConfigureAwait(false);
+            this.sysbenchDirectory = this.GetPackagePath(this.PackageName);
+
+            SysbenchOLTPState state = await this.stateManager.GetStateAsync<SysbenchOLTPState>($"{nameof(SysbenchOLTPState)}", cancellationToken)
+                ?? new SysbenchOLTPState();
+
+            if (!state.SysbenchInitialized)
+            {
+                // install sysbench using repo scripts
+                await this.InstallSysbenchOLTPPackage(cancellationToken).ConfigureAwait(false);
+                state.SysbenchInitialized = true;
+            }
+
+            // prepares database based on prepare arguments in profile 
+            this.sysbenchPrepareArguments = $@"oltp_common --tables={this.NumTables} --mysql-db={this.DatabaseName} --mysql-user=root prepare";
+
+            string sysbenchPath = this.PlatformSpecifics.Combine(this.sysbenchDirectory, SysbenchOLTPClientExecutor.SysbenchFileName);
+            await this.ExecuteCommandAsync<SysbenchOLTPClientExecutor>(sysbenchPath, this.sysbenchPrepareArguments, this.sysbenchDirectory, cancellationToken)
+                .ConfigureAwait(false);
+
+            await this.stateManager.SaveStateAsync<SysbenchOLTPState>($"{nameof(SysbenchOLTPState)}", state, cancellationToken);
+        }
+
+        private Task CaptureMetricsAsync(string results, DateTime startTime, DateTime endTime, EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                results.ThrowIfNullOrWhiteSpace(nameof(results));
+
+                this.Logger.LogMessage($"{nameof(SysbenchOLTPClientExecutor)}.CaptureMetrics", telemetryContext.Clone()
+                    .AddContext("results", results));
+
+                try
+                {
+                    SysbenchOLTPMetricsParser parser = new SysbenchOLTPMetricsParser(results);
+                    this.Logger.LogMetrics(
+                        toolName: "Sysbench",
+                        scenarioName: "OLTP",
+                        startTime,
+                        endTime,
+                        parser.Parse(),
+                        metricCategorization: "Sysbench",
+                        scenarioArguments: this.sysbenchExecutionArguments,
+                        this.Tags,
+                        telemetryContext);
+                }
+                catch (Exception exc)
+                {
+                    throw new WorkloadException($"Failed to parse sysbench output: {results}.", exc, ErrorReason.InvalidResults);
+                }
+            }
+
+            return Task.CompletedTask;
```

Comment:
This method doesn't have to be async and you don't have to return Task.CompletedTask